programming-extensions/programming-extension-processbuilder/src/main/java/org/objectweb/proactive/extensions/processbuilder/LinuxProcessBuilder.java

```diff
@@ -663,33 +664,60 @@ public OSLinuxProcess(Process p, OSUser user, String token, String scriptBaseFol
 
         @Override
         public void destroy() {
+            Process p = null;
             try {
                 LinuxProcessBuilder lpb = new LinuxProcessBuilder(user, null, scriptBaseFolder);
                 String killProcessTreeScript = new File(scriptBaseFolder,
                                                         PROCESS_TREE_KILLER_LOCATION).getAbsolutePath();
                 lpb.directory(new File(System.getProperty("java.io.tmpdir")));
                 lpb.command(killProcessTreeScript, this.cleanupTimeoutGetter.getCleanupTimeSecondsString());
                 lpb.environment().put("PROCESS_KILL_TOKEN", this.token);
-                Process p = lpb.start();
+                p = lpb.start();
                 p.waitFor();
 
                 if (logger.isDebugEnabled()) {
-                    BufferedReader br;
+                    printStdout(p, Level.DEBUG);
+                    printStderr(p, Level.DEBUG);
+                    logger.debug("token: " + this.token + " exitval:" + p.exitValue());
+                }
+            } catch (Exception e) {
+                printStderr(p, Level.WARN);
+                logger.warn("Failed to destroy OS process with token: " + this.token, e);
```

Comment:
I would rather do 
logger.warn("Failed to destroy OS process with token: " + this.token, e);
printStdout(p, Level.WARN);
printStderr(p, Level.WARN);